src/com/amazon/corretto/crypto/provider/EcParameters.java

```diff
@@ -0,0 +1,115 @@
+            // https://github.com/openjdk/jdk10u/blob/ef9178d7d8a4489640a31a1d0c88958724af5304/src/java.base/share/classes/sun/security/ssl/SupportedGroupsExtension.java#L188-L210
+            if (Utils.getJavaVersion() < 11) {
+                name = EcUtils.getOidFromName(name);
+            }
+        } else if (paramSpec instanceof ECGenParameterSpec) {
+            name = ((ECGenParameterSpec) paramSpec).getName();
+        } else if ("sun.security.util.ECKeySizeParameterSpec".equals(paramSpec.getClass().getName())) {
+            // OpenJDK's JCE sometimes passes ECKeySizeParameterSpec when initializing EC AlgorithmParameters. We can
+            // reference this class directly using instanceof up until Java 8, but in Java 9 it was moved to the
+            // java.base module and only exported to other JDK-internal modules. To work around this, we use reflection
+            // to get the key size and get the corresponding curve name from our own database. Future versions of java
+            // may restrict reflective access to private modules, so this functionality may break.
+            //
+            // https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/sun/security/util/ECParameters.java#L121
+            // https://github.com/corretto/corretto-11/blob/14c02261590b4dc01284888a7a51d39ff581ac8d/src/java.base/share/classes/module-info.java#L301-L314
+            try {
+                Method getKeySize = paramSpec.getClass().getMethod("getKeySize");
+                Integer keySize = Integer.class.cast(getKeySize.invoke(paramSpec));
+                name = EcUtils.getNameByKeySize(keySize);
+            } catch (ReflectiveOperationException e) {
+                // pass, perhaps due to reflective access control restrictions. rely null check below to throw.
+            }
+        }
+
+        if (name == null) {
+            throw new InvalidParameterSpecException("Only ECParameterSpec and ECGenParameterSpec supported");
+        }
+
+        ecInfo = EcUtils.getSpecByName(name);
+        if (ecInfo == null) {
+            throw new InvalidParameterSpecException("Unknown curve: " + paramSpec);
+        }
+    }
+
+    protected void engineInit(byte[] params) throws IOException {
+        String name = null;
+        try {
+            name = EcUtils.getNameByEncoded(params);
+        } catch (RuntimeCryptoException e) {
+            // pass, handle via null check below
+        }
+        if (name == null) {
+            throw new IOException("Only named EcParameters supported");
+        }
+        ecInfo = EcUtils.getSpecByName(name);
+        if (ecInfo == null) {
+            throw new IOException("Unknown named curve: " + name);
+        }
+    }
+
+    protected void engineInit(byte[] params, String unused) throws IOException {
+        engineInit(params);
```

Comment:
Is this the desired behavior? Alternative is to throw something like `UnsupportedOperationException`. Just checking.

Reply:
this is intentional, and is intended to preserve behavioral compatibility with [SunJCE's implementation](https://github.com/corretto/corretto-11/blob/develop/src/java.base/share/classes/sun/security/util/ECParameters.java#L191).